### PR TITLE
Fix `Except` not working with generic properties

### LINKED ISSUE #232: Except is broken when used with generic property
```ts
class Example {
    foo: number;
    bar: string;

    constructor() {
        this.foo = 1;
        this.bar = '';
    }

    [key: string]: unknown;
}

type Except<ObjectType, KeysType extends keyof ObjectType> = Pick<ObjectType, Exclude<keyof ObjectType, KeysType>>;
const test: Except<Example, 'bar'> = new Example();

// `foo` is now `unknown`
console.log(test.foo + 5);
```

[Playground link](https://www.typescriptlang.org/play?ts=4.4.0-beta#code/MYGwhgzhAECiAeYC2AHEBTaBvAUNf0AZgPbEBc0AdgK5IBG6ATgNx4F1iMUQAujAlpQDmrNvmDFKvRtWA9ijABQBKbGIL4eAC34QAdCWLQAvNACMrDRu269HRiegByJ5YIBfHOoDaAa3QAntx8gkIAuhTUlL6UxADulKyeODwBKJgIwOgoPAA8APJ0AFbocgAqaegANNAA0oEQFenQ6PA86JQAJjD+AcSE0IUl5ZUAfI4ACvzAvgXFpTxN1XDwoNSd6Lm9-YPzI+k19QGNY6OsElI80O28FJnZeQjIaMtO9k7jppTocSvPGCpRAB6IHQAAGhjB0F0VHi4KiMXilDBOAuEGIGD0IGIQkUNx4BlI0AA1NAAKzKZhAA)

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #232

### CODE REVIEW

File: source/except.d.ts [hunk 1]
```diff
@@ -21,4 +21,12 @@ type FooWithoutA = Except<Foo, 'a' | 'c'>;
 
 @category Utilities
 */
-export type Except<ObjectType, KeysType extends keyof ObjectType> = Pick<ObjectType, Exclude<keyof ObjectType, KeysType>>;
+
+type EqualsTest<T> = <A>() => A extends T ? 1 : 0;
+type Equals<A1, A2> = EqualsTest<A2> extends EqualsTest<A1> ? 1 : 0;
```
Comment (szmarczak) [inline source/except.d.ts:26]: Should this be a separate type? E.g. `EqualsStrict`?

Comment (szmarczak) [reply source/except.d.ts]: Simplified

Comment (sindresorhus) [reply source/except.d.ts:26]: Looks like this type: https://github.com/sindresorhus/type-fest/blob/70fe5242645071808bc556e8aaa0d2ba9e12991b/source/includes.d.ts#L1-L10

Can you move that one out into `source/internal` and use it for now. We can consider exposing it publicly later on.

Comment (sindresorhus) [reply source/except.d.ts:26]: Any reason you're using `1` and `0` here instead of `true` and `false`?

File: source/except.d.ts [hunk 2]
```diff
@@ -21,4 +21,12 @@ type FooWithoutA = Except<Foo, 'a' | 'c'>;
...
+
+type Filter<K, I> = Equals<K, I> extends 1 ? never : (K extends I ? never : K);
```
Comment (sindresorhus) [inline source/except.d.ts:28]: Needs a doc comment.

File: source/except.d.ts [hunk 3]
```diff
@@ -21,4 +21,12 @@ type FooWithoutA = Except<Foo, 'a' | 'c'>;
...
+
+export type Except<T, I> = {
```
Comment (sindresorhus) [inline source/except.d.ts:30]: Descriptive generic type names.

File: source/except.d.ts (hunk 1 above)
Comment (sindresorhus) [reply source/except.d.ts:26]: Also link to https://stackoverflow.com/a/68963796/64949 for explanation of the type.

**Comment by sindresorhus** [comment]
I think you missed some of my comments in https://github.com/sindresorhus/type-fest/pull/263#discussion_r706084325

File: source/except.d.ts [hunk 4]
```diff
@@ -1,3 +1,31 @@
+import {IsEqual} from './includes';
+
+/**
+Returns `never` if `Exclude` is strictly equal to `Key`.
```
Comment (sindresorhus) [inline source/except.d.ts:4]: A description of the purpose of the type would be useful before the technical explanation.

File: source/except.d.ts [hunk 5]
```diff
@@ -1,3 +1,31 @@
...
+//=> never
+```
+
+@example
+```
+type Filtered = Filter<'bar', 'foo'>;
+//=> 'bar'
+```
+*/
+
```
Comment (sindresorhus) [inline source/except.d.ts:26]:
```suggestion
```

File: source/except.d.ts (hunk 4 above)
Comment (szmarczak) [reply source/except.d.ts:6]: Done.

**Comment by xiaoyu-tamu** [comment]
@szmarczak the editor autocomplete feature is disappeared after this commit 

#267

**Pull request merged**